File: flax_search_service/docs/tutorial/php/t5_get_doc.php
<?php
include_once "flax/flaxclient.php";

# open a connection to Flax Server
$conn = new FlaxSearchService('http://localhost:8080');

# get a reference to the database
$db = $conn->getDatabase('books');

# fetch and print the document
print_r($db->getDocument($argv[1]));

?>

File: flax_search_service/docs/tutorial/perl/t5_get_doc.pl
#!/usr/bin/perl

use strict;
use warnings;
use Error qw(:try);

use Flax;
use Flax::Error;
use Flax::Field;
use Flax::SummarySettings;

my $flax = Flax->new('url'=>'http://localhost:8080');
my $db = $flax->getDatabase('books');

# fetch and print the document
my $doc = $db->getDocument($ARGV[0]);
my $k;
my $v;
while (($k, $v) = each (%{$doc}))
{
    print "$k: ". join(', ', @$v) ."\n";
}
